grapple/types/redirects.py

```diff
@@ -35,11 +36,26 @@ def resolve_old_url(self, info, **kwargs) -> str:
 
     def resolve_new_url(self, info, **kwargs) -> Optional[str]:
         """
-        Resolve the value of `new_url`. If `redirect_page` is specified then its
-        URL is prioritised.
+        Resolve the value of `new_url`. If `redirect_page` is specified then
+        `link` is used. Otherwise, ensure that the redirect link is absolute.
         """
+        if self.redirect_page:
+            return self.link  # Handled by the `Redirect` model
+
+        elif self.redirect_link:
+            parsed_url = urlparse(self.redirect_link)
+
+            if not parsed_url.scheme:  # url without scheme is not absolute
+                return (
+                    self.site.root_url.rstrip("/")
+                    + "/"
+                    + self.redirect_link.lstrip("/")
+                )
+            else:
+                return self.redirect_link
 
-        return self.link
+        else:
+            return None
```

Comment:
Just got time to look at this properly.

`Redirect.redirect_link` is an URLField and will always have a URL scheme and netloc (or at least it should), so this change doesn't improve the behaviour whatsoever.

Looking at #384, the thought process was that `newUrl` should always be a full URL to reduce ambiguity in a multi-site setup. As far as I can unpick the results from #384 the relative URL comes from the fact that the redirect applies to all sites. 

Now, if we change to return full URLs, we should document it properly in the changelog and release notes. I am not opposed to always returning the full URL, but definitely want to have proper test coverage for redirect that apply to all sites

Reply:
saying that, I do wonder if we're adding maintenance burden as we'll need to replicate https://github.com/wagtail/wagtail/blob/a09bba67cd58f519f3ae5bff32575e7ce9244031/wagtail/contrib/redirects/models.py#L68-L81

Reply:
From what I can tell, `newUrl` being relative should only occur in single site setups, as we call into `Page.url` which returns relative URLs for single site, or an absolute URL if there are multiple sites. I'm having a hard time coming up with ways that this would be problematic. I think the original issue is erroneous.